<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ChatClientWidget</class>
 <widget class="QWidget" name="ChatClientWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1047</width>
    <height>701</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QWidget{background-color: rgb(59,64,86);}
</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QStackedWidget" name="mainPages">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="connection">
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,1,0,0">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer_4">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QLabel" name="title">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="styleSheet">
            <string notr="true">QLabel{font: bold 108pt 'Calibri'; font-weight:900;}</string>
           </property>
           <property name="text">
            <string>&lt;span style=&quot;color:#39A7DE&quot;&gt;Chat&lt;/span&gt;&lt;span style=&quot;color:white&quot;&gt;Client&lt;/span&gt;</string>
           </property>
           <property name="textFormat">
            <enum>Qt::RichText</enum>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_3">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item alignment="Qt::AlignTop">
          <widget class="QPushButton" name="showLogButton0">
           <property name="focusPolicy">
            <enum>Qt::ClickFocus</enum>
           </property>
           <property name="styleSheet">
            <string notr="true">QPushButton{background-color: #39A7DE; color:white; border-radius:3%;font: bold 16pt 'Calibri'; font-weight:700;  border-style: outset; padding: 0 3px 0 3px}
QPushButton:hover{background-color: #2997CE; border-style: outset;}
QPushButton:pressed{background-color: #1987BE;  border-style: inset;}</string>
           </property>
           <property name="text">
            <string>Log</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,5,4">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <spacer name="horizontalSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QFrame" name="frame">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Ignored" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="styleSheet">
            <string notr="true">QWidget{background-color: white;}
QFrame{border-radius: 10%}
QProgressBar{ border-radius: 0px;}
QLineEdit{border-radius:0; color: #404040;font: 12pt 'Calibri'; font-weight:500; border: 1px solid #A0A0A0; border-style: inset;}
QLabel{color: #606060; font: bold 14pt 'Calibri'; font-weight:700;}
QPushButton{background-color: #39A7DE; color:white; border-radius:3%;font: bold 16pt 'Calibri'; font-weight:700;  border-style: outset;}
QPushButton:hover{background-color: #2997CE; border-style: outset;}
QPushButton:pressed{background-color: #1987BE;  border-style: inset;}
QPushButton:disabled{background-color: #404040; color: lightgrey;}</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QStackedWidget" name="connectionPages">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <widget class="QWidget" name="serverSelection">
               <layout class="QVBoxLayout" name="verticalLayout_4">
                <property name="spacing">
                 <number>6</number>
                </property>
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="serverAddressLabel">
                  <property name="text">
                   <string>Server Address</string>
                  </property>
                  <property name="buddy">
                   <cstring>serverAddressInput</cstring>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="serverAddressInput">
                  <property name="focusPolicy">
                   <enum>Qt::ClickFocus</enum>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="connectButton">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="focusPolicy">
                   <enum>Qt::ClickFocus</enum>
                  </property>
                  <property name="text">
                   <string>Connect</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_4">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="login">
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item>
                 <widget class="QLabel" name="participantNameLabel">
                  <property name="font">
                   <font>
                    <family>Calibri</family>
                    <pointsize>14</pointsize>
                    <weight>87</weight>
                    <italic>false</italic>
                    <bold>true</bold>
                   </font>
                  </property>
                  <property name="text">
                   <string>Participant Name</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                  </property>
                  <property name="buddy">
                   <cstring>participantNameInput</cstring>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QLineEdit" name="participantNameInput">
                  <property name="sizePolicy">
                   <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                    <horstretch>0</horstretch>
                    <verstretch>0</verstretch>
                   </sizepolicy>
                  </property>
                  <property name="focusPolicy">
                   <enum>Qt::ClickFocus</enum>
                  </property>
                  <property name="styleSheet">
                   <string notr="true"/>
                  </property>
                  <property name="text">
                   <string/>
                  </property>
                  <property name="maxLength">
                   <number>15</number>
                  </property>
                  <property name="frame">
                   <bool>true</bool>
                  </property>
                  <property name="placeholderText">
                   <string/>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="loginButton">
                  <property name="focusPolicy">
                   <enum>Qt::ClickFocus</enum>
                  </property>
                  <property name="text">
                   <string>Login</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeType">
                   <enum>QSizePolicy::Expanding</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
                <item>
                 <widget class="QPushButton" name="backButton">
                  <property name="focusPolicy">
                   <enum>Qt::ClickFocus</enum>
                  </property>
                  <property name="text">
                   <string>Back</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="errorLabel">
              <property name="styleSheet">
               <string notr="true">QLabel{color: red; font: 10pt 'Calibri';}</string>
              </property>
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QProgressBar" name="connectProgressBar">
              <property name="styleSheet">
               <string notr="true"/>
              </property>
              <property name="maximum">
               <number>4</number>
              </property>
              <property name="value">
               <number>0</number>
              </property>
              <property name="format">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="quitButton">
              <property name="focusPolicy">
               <enum>Qt::ClickFocus</enum>
              </property>
              <property name="text">
               <string>Quit</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="chat">
      <property name="styleSheet">
       <string notr="true">QWidget{background-color: rgb(59,64,86);}
.QFrame{border-radius:5%;}
QTextEdit, QLineEdit, QListWidget, .QFrame, QLabel{background-color: white;}
QTextEdit { color: #404040;font: 12pt 'Calibri'; font-weight:500; border: 1px solid #A0A0A0; border-style: inset;}
QLabel{color: #606060; font: bold 14pt 'Calibri'; font-weight:700; padding: 0 2px 0 2px}
QPushButton{background-color: #39A7DE; color:white; border-radius:3%;font: bold 16pt 'Calibri'; font-weight:700;  border-style: outset; padding: 0 3px 0 3px;}
QPushButton:hover{background-color: #2997CE; border-style: outset;}
QPushButton:pressed{background-color: #1987BE;  border-style: inset;}</string>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="leftMargin">
        <number>5</number>
       </property>
       <property name="topMargin">
        <number>5</number>
       </property>
       <property name="rightMargin">
        <number>5</number>
       </property>
       <property name="bottomMargin">
        <number>5</number>
       </property>
       <property name="spacing">
        <number>5</number>
       </property>
       <item row="2" column="1">
        <layout class="QVBoxLayout" name="verticalLayout_7">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="yourNicknameLabel">
             <property name="text">
              <string>Your nickname:</string>
             </property>
             <property name="buddy">
              <cstring>nicknameInput</cstring>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="nicknameInput">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QListWidget" name="participantsList"/>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="QFrame" name="frame1">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_6">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>2</number>
          </property>
          <property name="topMargin">
           <number>2</number>
          </property>
          <property name="rightMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="participantsLabel">
            <property name="text">
             <string>Participants</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QFrame" name="frame0">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <property name="spacing">
           <number>0</number>
          </property>
          <property name="leftMargin">
           <number>2</number>
          </property>
          <property name="topMargin">
           <number>2</number>
          </property>
          <property name="rightMargin">
           <number>2</number>
          </property>
          <property name="bottomMargin">
           <number>2</number>
          </property>
          <item>
           <widget class="QLabel" name="chatLabel">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>Chat</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <item>
          <widget class="QTextEdit" name="chatBox">
           <property name="readOnly">
            <bool>true</bool>
           </property>
           <property name="html">
            <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Calibri'; font-size:12pt; font-weight:496; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:10pt; font-weight:400;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="chatInput">
           <property name="placeholderText">
            <string>Type a message</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="3" column="0">
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QPushButton" name="optionsButton">
           <property name="text">
            <string>Options</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="showLogButton1">
           <property name="text">
            <string>Log</string>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="horizontalSpacer_5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </item>
       <item row="3" column="1">
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <spacer name="horizontalSpacer_6">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>40</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <widget class="QPushButton" name="disconnectButton">
           <property name="text">
            <string>Disconnect</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>serverAddressInput</tabstop>
  <tabstop>connectButton</tabstop>
  <tabstop>quitButton</tabstop>
  <tabstop>showLogButton0</tabstop>
  <tabstop>participantNameInput</tabstop>
  <tabstop>loginButton</tabstop>
  <tabstop>backButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
